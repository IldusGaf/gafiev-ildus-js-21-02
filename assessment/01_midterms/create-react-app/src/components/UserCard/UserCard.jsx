import React from 'react';
import './UserCard.css';
import { FormOutlined } from '@ant-design/icons';
import { useTranslation } from 'react-i18next';
import '../../locale/i18next';

const UserCard = (props) => {
  const { t } = useTranslation();
  return (
    <div className={`user__card-container ${props.darkTheme && 'user_dark'}`}>
      <figure>
        <img src={props.picture} alt="Load Error" />
      </figure>
      <div className="user__data">
        <div className="user__info">
          <h2 className={props.darkTheme && 'user_dark'}>
            {props.title
              ? `${props.title} ${props.firstName} ${props.lastName}`
              : `${props.firstName} ${props.lastName}`}
          </h2>
          <div className="info__line">
            <span className="info__title">{t('userCard.gender')}: </span>
            <span>{props.gender}</span>
          </div>
          <div className="info__line">
            <span className="info__title">{t('userCard.dateOfBirth')}: </span>
            <span> {props.dateOfBirth && props.dateOfBirth.slice(0, 10)}</span>
          </div>
          <div className="info__line">
            <span className="info__title">{t('userCard.registerDate')}: </span>
            <span>{props.registerDate && props.registerDate.slice(0, 10)}</span>
          </div>
          <div className="info__line">
            <span className="info__title">E-mail: </span>
            <span>{props.email}</span>
          </div>
          <div className="info__line">
            <span className="info__title">{t('userCard.phone')}: </span>
            <span>{props.phone}</span>
          </div>
        </div>
        <div className="user__id">
          <span className="info__title">ID: </span>
          <span>{props.id}</span>
        </div>
      </div>
      <div className={`user__edit ${props.authId === props.id && 'user__edit_visible'}`} onClick={props.edit}>
        <FormOutlined />
        <span className="edit__text">{t('userCard.edit')}</span>
      </div>
    </div>
  );
};

export default UserCard;
