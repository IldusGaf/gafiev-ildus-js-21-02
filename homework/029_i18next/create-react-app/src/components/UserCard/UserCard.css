.img-sizes {
  width: 250px;
  height: 250px;
}
.user__card-container {
  width: 100%;
  padding: 20px;
  background: #ffff;
  border-radius: 20px;
  display: flex;
  gap: 20px;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: center;
  position: relative;
}
.user__card-container figure {
  width: 250px;
  height: 250px;
  border-radius: 16px;
}
@media (max-width: 800px) {
  .user__card-container figure {
    width: 180px;
    height: 180px;
  }
}
.user__card-container figure img {
  width: 100%;
  height: 100%;
}
.user__card-container .user__data {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}
.user__card-container .user__data .info__title {
  font-weight: bold;
}
.user__card-container .user__data .user__info h2 {
  font-size: 25px;
  margin-bottom: 20px;
}
@media (max-width: 800px) {
  .user__card-container .user__data .user__info h2 {
    font-size: 14px;
  }
}
.user__card-container .user__data .info__line {
  font-size: 14px;
  margin-bottom: 10px;
}
@media (max-width: 800px) {
  .user__card-container .user__data .info__line {
    font-size: 12px;
  }
}
.user__card-container .user__data .user__id {
  font-size: 14px;
}
@media (max-width: 800px) {
  .user__card-container .user__data .user__id {
    font-size: 12px;
  }
}
.user__card-container .user__edit {
  position: absolute;
  top: 20px;
  right: 20px;
  font-size: 12px;
  display: none;
}
.user__card-container .user__edit_visible {
  display: block;
}
.user_dark {
  background: #001529;
  color: #ffff;
}
@media (max-width: 800px) {
  .edit__text {
    display: none;
  }
}
/*# sourceMappingURL=UserCard.css.map */