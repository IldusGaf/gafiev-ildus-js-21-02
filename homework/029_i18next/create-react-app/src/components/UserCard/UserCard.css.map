{"version":3,"sources":["UserCard.less"],"names":[],"mappings":"AASA;EACI,YAAA;EACA,aAAA;;AAGJ;EACI,WAAA;EAEA,aAAA;EACA,iBAAA;EACA,mBAAA;EAEA,aAAA;EACA,SAAA;EACA,eAAA;EACA,mBAAA;EACA,uBAAA;EAEA,kBAAA;;AAbJ,qBAeI;EAnBA,YAAA;EACA,aAAA;EAoBI,mBAAA;;AAEA,QAA0B;EAA1B,qBAJJ;IAKQ,YAAA;IACA,aAAA;;;AArBZ,qBAeI,OASI;EACI,WAAA;EACA,YAAA;;AA1BZ,qBA8BI;EACI,YAAA;EACA,aAAA;EACA,sBAAA;EACA,8BAAA;;AAlCR,qBA8BI,YAMI;EACI,iBAAA;;AArCZ,qBA8BI,YAUI,YAAY;EACR,eAAA;EACA,mBAAA;;AAEA,QAA0B;EAA1B,qBAdR,YAUI,YAAY;IAKJ,eAAA;;;AA7ChB,qBA8BI,YAmBI;EACI,eAAA;EACA,mBAAA;;AAEA,QAA0B;EAA1B,qBAvBR,YAmBI;IAKQ,eAAA;;;AAtDhB,qBA8BI,YA4BI;EACI,eAAA;;AAEA,QAA0B;EAA1B,qBA/BR,YA4BI;IAIQ,eAAA;;;AA9DhB,qBAmEI;EACI,kBAAA;EACA,SAAA;EACA,WAAA;EACA,eAAA;EACA,aAAA;;AAxER,qBA2EI;EACI,cAAA;;AAIR;EACI,mBAAA;EACA,YAAA;;AAGJ,QAA0B;EACtB;IACI,aAAA","file":"UserCard.css"}