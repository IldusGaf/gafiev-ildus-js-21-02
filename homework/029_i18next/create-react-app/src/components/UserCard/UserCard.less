@card-height: 300px;
@card-padding: 20px;
@container-gap: 20px;
@text-gap: 10px;
@name-font-size: 25px;
@font-size: 14px;
@edit-font-size: 12px;
@card-radius: 20px;
@picture-radius: 16px;
.img-sizes {
    width: 250px;
    height: 250px;
}

.user__card-container {
    width: 100%;
    //height: @card-height;
    padding: @card-padding;
    background: #ffff;
    border-radius: @card-radius;

    display: flex;
    gap: @container-gap;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: center;

    position: relative;

    figure {
        .img-sizes;
        border-radius: @picture-radius;

        @media (max-width: 800px) {
            width: 180px;
            height: 180px;
        }

        img {
            width: 100%;
            height: 100%;
        }
    }

    .user__data {
        height: 100%;
        display: flex;
        flex-direction: column;
        justify-content: space-between;

        .info__title {
            font-weight: bold;
        }

        .user__info h2 {
            font-size: @name-font-size;
            margin-bottom: @container-gap;

            @media (max-width: 800px) {
                font-size: @font-size;
            }
        }

        .info__line {
            font-size: @font-size;
            margin-bottom: @text-gap;

            @media (max-width: 800px) {
                font-size: @edit-font-size;
            }
        }

        .user__id {
            font-size: @font-size;

            @media (max-width: 800px) {
                font-size: @edit-font-size;
            }
        }
    }

    .user__edit {
        position: absolute;
        top: @container-gap;
        right: @container-gap;
        font-size: @edit-font-size;
        display: none;
    }

    .user__edit_visible {
        display: block;
    }
}

.user_dark {
    background: #001529;
    color: #ffff;
}

@media (max-width: 800px) {
    .edit__text {
        display: none;
    }
}