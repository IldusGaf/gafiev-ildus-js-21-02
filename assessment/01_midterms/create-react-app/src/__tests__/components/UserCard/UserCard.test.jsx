import React from "react";
import { shallow } from 'enzyme';
import UserCard from "../../../components/UserCard/UserCard";

describe('UserCard component', () => {
    test('should render component', () => {
        const wrapper = shallow(<UserCard />);
        expect(wrapper.find('div.user__container')).toHaveLength(1);
    });

    test('snapshot test', () => {
        const wrapper = shallow(<UserCard />);
        expect(wrapper).toMatchSnapshot();
      });

    test('should render data', () => {
        const id = '216544654697498';
        const title = 'mr';
        const firstName = 'James';
        const lastName = 'Bond';
        const dateOfBirth = '12.06.1989';
        const registerDate = '13.12.2020';
        const gender = 'male';
        const email = 'ex@ex.com';
        const phone = '12315465656';
        const picture = 'pictureUrl';
        const wrapper = shallow(<UserCard
           title={title}
           firstName={firstName} 
           lastName={lastName}
           dateOfBirth={dateOfBirth}
           registerDate={registerDate}
           picture={picture}
           id={id}
           email={email}
           phone={phone}
           gender={gender}
        />);
        expect(wrapper.find('div.user__info').find('h2').text()).toBe(`${title} ${firstName} ${lastName}`);
        expect(wrapper.find('div.user__id').find('span').last().text()).toBe(id);
        expect(wrapper.find('figure').find('img').prop('src')).toBe(picture);
        expect(wrapper.find('div.gender').find('span').last().text()).toBe(gender);
        expect(wrapper.find('div.birthday').find('span').last().text()).toBe(dateOfBirth);
        expect(wrapper.find('div.regday').find('span').last().text()).toBe(registerDate);
        expect(wrapper.find('div.email').find('span').last().text()).toBe(email);
        expect(wrapper.find('div.phone').find('span').last().text()).toBe(phone);
    });
});
