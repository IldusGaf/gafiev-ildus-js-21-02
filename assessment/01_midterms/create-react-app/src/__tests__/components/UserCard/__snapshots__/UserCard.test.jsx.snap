// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component snapshot test 1`] = `
<div
  className="user__card-container undefined"
>
  <figure>
    <img
      alt="Load Error"
    />
  </figure>
  <div
    className="user__data"
  >
    <div
      className="user__info"
    >
      <h2>
        undefined undefined
      </h2>
      <div
        className="info__line"
      >
        <span
          className="info__title"
        >
          Gender
          : 
        </span>
        <span />
      </div>
      <div
        className="info__line"
      >
        <span
          className="info__title"
        >
          Date of birth
          : 
        </span>
        <span>
           
        </span>
      </div>
      <div
        className="info__line"
      >
        <span
          className="info__title"
        >
          Register date
          : 
        </span>
        <span />
      </div>
      <div
        className="info__line"
      >
        <span
          className="info__title"
        >
          E-mail: 
        </span>
        <span />
      </div>
      <div
        className="info__line"
      >
        <span
          className="info__title"
        >
          Phone
          : 
        </span>
        <span />
      </div>
    </div>
    <div
      className="user__id"
    >
      <span
        className="info__title"
      >
        ID: 
      </span>
      <span />
    </div>
  </div>
  <div
    className="user__edit user__edit_visible"
  >
    <ForwardRef(FormOutlined) />
    <span
      className="edit__text"
    >
      Edit
    </span>
  </div>
</div>
`;
